### PR TITLE
Make testwrap python script

### PR DESCRIPTION
**Pull request by nbyavuz**

### CODE REVIEW

File: src/tools/testwrap [hunk 1]
```diff
@@ -1,25 +1,33 @@
...
+print('# executing test in {} group {} test {}, builddir {}'
+  .format(testdir, args.testgroup, args.test_name, args.builddir))
+
+if os.path.exists(testdir) and os.path.isdir(testdir):
+  shutil.rmtree(testdir)
+os.makedirs(testdir)
+
+os.chdir(args.srcdir)
+
+env_dict = dict(TESTOUTDIR=testdir, TESTDIR=args.builddir, **os.environ)
```
Comment (anarazel) [inline src/tools/testwrap:32]: It's much more common to initialize dicts with {} in python... And it seems not quite right to allow os.environ to overwrite TESTOUTDIR etc?

File: meson.build [hunk 2]
```diff
@@ -2678,11 +2678,11 @@ foreach test: tests
...
-        t['sd'],
-        meson.build_root(),
-        t['bd'],
-        t['name'],
-        'pg_regress',
+        '--srcdir', t['sd'],
+        '--basedir', meson.build_root(),
+        '--builddir', t['bd'],
+        '--testgroup', t['name'],
+        '--test_name', 'pg_regress',
```
Comment (anarazel) [inline meson.build:2685]: Looks a bit odd to have 'srcdir' etc (i.e. words concatenated together without separator), but then 'test_name' (i.e. concatenated with `_` as separator).

File: meson.build [hunk 3]
```diff
@@ -2692,7 +2692,7 @@ foreach test: tests
     endif
 
     test(t['name'] / 'pg_regress',
-      sh,
```
Comment (anarazel) [inline meson.build:2695]: I think this will have removed the last 'sh' user, right? If so, let's remove it.

**Review by anarazel — APPROVED** [review]
Except for that one comment about 'sh', feel free to merge!

File: meson.build (hunk 3 above)
Comment (nbyavuz) [reply meson.build]: Yes, it was last. Removing and merging.

**Comment by nbyavuz** [comment]
Merged.

**Pull request merged**